### PR TITLE
feat(vscode) : new app

### PR DESCRIPTION
**Pull request by floriansipp**
As the title says, only question left is for apps with a - something do we still add -PKG-REL or not ? What do you think @greut ?

### CODE REVIEW

File: apps/vscode/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+WORKDIR /apps/${APP_NAME}
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update -qy && \
+    apt-get install --no-install-recommends -qy ca-certificates wget gpg binutils && \
+    wget -qO- https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > packages.microsoft.gpg && \
+    install -D -o root -g root -m 644 packages.microsoft.gpg /etc/apt/keyrings/packages.microsoft.gpg && \
```
Comment (greut) [inline apps/vscode/Dockerfile:16]: I think this isn't required as the `.list` points to the key explicitly.

**Comment by greut** [comment]
@floriansipp I'd keep the pkgrel and you may try to get rid of the upstream `-12345...` by using `apt-cache madison`, see: https://github.com/HIP-infrastructure/app-in-browser/blob/dev/services/base-images/virtualgl/Dockerfile#L23

File: apps/vscode/Dockerfile (hunk 1 above)
Comment (floriansipp) [reply apps/vscode/Dockerfile:16]: It does not work without this

**Comment by floriansipp** [comment]
> @floriansipp I'd keep the pkgrel and you may try to get rid of the upstream `-12345...` by using `apt-cache madison`, see: https://github.com/HIP-infrastructure/app-in-browser/blob/dev/services/base-images/virtualgl/Dockerfile#L23

Indeed, that seems to do the trick

**Comment by greut** [comment]
it crashed on my first, but then made it to this stage. VS managed to install the extension, but no Python in the image so it's a sad panda.

Capture d’écran 2024-08-26 à 13 18 28

The crash.

<details>
<summary>Crash log</summary>

```
/docker-entrypoint.sh: Configuration complete; ready for start up
Running  as chorus on CPU...
[22:0826/111404.252738:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[22:0826/111404.512758:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[22:0826/111404.512818:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[22:0826/111404.514911:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111404.514947:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111404.514957:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111404.514963:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111404.517307:ERROR:zygote_host_impl_linux.cc(270)] Failed to adjust OOM score of renderer with pid 52: Permission denied (13)
[22:0826/111404.708068:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[22:0826/111404.708290:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[22:0826/111404.708492:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
glx: failed to create drisw screen
[52:0826/111404.878217:ERROR:viz_main_impl.cc(198)] Exiting GPU process due to errors during initialization
[22:0826/111404.922548:ERROR:zygote_host_impl_linux.cc(270)] Failed to adjust OOM score of renderer with pid 84: Permission denied (13)
[main 2024-08-26T11:14:04.945Z] update#setState idle
[22:0826/111404.951596:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111405.063351:ERROR:zygote_host_impl_linux.cc(270)] Failed to adjust OOM score of renderer with pid 105: Permission denied (13)
MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
glx: failed to create drisw screen
[105:0826/111405.296716:ERROR:viz_main_impl.cc(198)] Exiting GPU process due to errors during initialization
[22:0826/111405.302970:ERROR:zygote_host_impl_linux.cc(270)] Failed to adjust OOM score of renderer with pid 117: Permission denied (13)
[84:0826/111405.384577:ERROR:command_buffer_proxy_impl.cc(131)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.
[22:0826/111406.650527:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111407.531307:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111407.775741:ERROR:zygote_host_impl_linux.cc(270)] Failed to adjust OOM score of renderer with pid 176: Permission denied (13)
[22:0826/111408.775049:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111409.385700:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111413.877281:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111418.342212:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")

(code:22): dconf-WARNING **: 11:14:29.335: failed to commit changes to dconf: Unknown or unsupported transport ?disabled? for address ?disabled:?

(code:22): Gdk-WARNING **: 11:14:29.380: The program 'code' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadShmSeg (invalid shared segment parameter)'.
  (Details: serial 1001 error_code 128 request_code 130 (MIT-SHM) minor_code 3)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)

(code:22): dconf-WARNING **: 11:17:18.862: failed to commit changes to dconf: Unknown or unsupported transport ?disabled? for address ?disabled:?

(code:22): dconf-WARNING **: 11:17:20.318: failed to commit changes to dconf: Unknown or unsupported transport ?disabled? for address ?disabled:?

(code:22): dconf-WARNING **: 11:17:20.334: failed to commit changes to dconf: Unknown or unsupported transport ?disabled? for address ?disabled:?

(code:22): dconf-WARNING **: 11:17:20.350: failed to commit changes to dconf: Unknown or unsupported transport ?disabled? for address ?disabled:?

(code:22): dconf-WARNING **: 11:17:20.377: failed to commit changes to dconf: Unknown or unsupported transport ?disabled? for address ?disabled:?

(code:22): dconf-WARNING **: 11:17:20.377: failed to commit changes to dconf: Unknown or unsupported transport ?disabled? for address ?disabled:?
[22:0826/111720.711662:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111725.892990:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111726.395684:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111731.971958:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[22:0826/111733.827361:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
```

</details>

**Review by greut — APPROVED** [review]

**Pull request merged**